genai/embed.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genai
+
+import (
+	"context"
+	"fmt"
+
+	pb "cloud.google.com/go/ai/generativelanguage/apiv1beta/generativelanguagepb"
+)
+
+func (c *Client) EmbeddingModel(name string) *EmbeddingModel {
+	return &EmbeddingModel{
+		c:        c,
+		name:     name,
+		fullName: fmt.Sprintf("models/%s", name),
+	}
+}
+
+// EmbeddingModel is a model that computes embeddings.
+// Create one with [Client.EmbeddingModel].
+type EmbeddingModel struct {
+	c        *Client
+	name     string
+	fullName string
+	TaskType TaskType
+}
+
+func (m *EmbeddingModel) EmbedContent(ctx context.Context, parts ...Part) (*EmbedContentResponse, error) {
+	return m.EmbedContentTitle(ctx, "", parts...)
+}
+
+func (m *EmbeddingModel) EmbedContentTitle(ctx context.Context, title string, parts ...Part) (*EmbedContentResponse, error) {
```

Comment:
Nit: maybe `EmbedContentWithTitle` sounds a bit better? `EmbedContentTitle` sounds like we're embedding something that's a "content title" and it's not clear what that is.

Also add docstrings on these `Embed*` methods

Reply:
Done.